src/handler/patch.ts

```diff
@@ -0,0 +1,61 @@
+import 'dotenv/config';
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import logger from '../util/logger';
+import { addHttpHeaders } from '../util/httpHeaders';
+import {
+  archiveOldCreateCurrentRecord,
+  getBySystemNumberAndCreatedTimestamp,
+} from '../services/database';
+import { validateUpdateVinRequest, validateVins } from '../validators/patch';
+import { processPatchVinRequest } from '../processors/processPatchVinRequest';
+import { TechrecordGet } from '../models/post';
+import { formatTechRecord } from '../util/formatTechRecord';
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Amend VIN Called');
+
+  const isRequestInvalid: APIGatewayProxyResult | undefined = validateUpdateVinRequest(event);
+
+  if (isRequestInvalid) {
+    return isRequestInvalid;
+  }
+
+  logger.debug('Request is Valid');
+
+  const systemNumber: string = decodeURIComponent(event.pathParameters?.systemNumber as string);
+  const createdTimestamp: string = decodeURIComponent(event.pathParameters?.createdTimestamp as string);
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  const newVin: string = JSON.parse(event.body!).newVin! as string;
+
+  const currentRecord: TechrecordGet = await getBySystemNumberAndCreatedTimestamp(
+    systemNumber,
+    createdTimestamp,
+  ) as TechrecordGet;
+
+  const isVinInvalid: APIGatewayProxyResult | undefined = validateVins(currentRecord, newVin.toUpperCase());
```

Comment:
How are you handling the undefined that might come back from this method?

Reply:
If the function is undefined then the handler moves on. If its defined it stops and returns the error. If that makes sense?

Reply:
Yeah this is the same as how all the others work, they just set the HTTP headers in the if rather than inside the function